### PR TITLE
Closes #3345 - implement WeMo Coffee Maker

### LINKED ISSUE #3345: [WIP] Add support for CoffeMaker to WeMo Binding
See https://community.openhab.org/t/wemo-coffe-maker/12667/8 and
https://github.com/openhab/openhab1-addons/issues/4275
Testing JAR can be found [here](https://drive.google.com/open?id=0BxdTuQI5IDvwTFppUUUzY0E0REk)

### PR DESCRIPTION
**Pull request by hmerk**
Signed-off-by: Hans-Jörg Merk <hans-joerg.merk@t-online.de>

### CODE REVIEW

File: extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/handler/WemoCoffeeHandler.java [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+                                + "&lt;attribute&gt;&lt;name&gt;Cleaning&lt;/name&gt;&lt;value&gt;NULL&lt;/value&gt;&lt;/attribute&gt;</attributeList>"
+                                + "</u:SetAttributes>" + "</s:Body>" + "</s:Envelope>";
+
+                        String wemoURL = getWemoURL("deviceevent");
+
+                        if (wemoURL != null) {
+                            String wemoCallResponse = WemoHttpCall.executeCall(wemoURL, soapHeader, content);
+                            if (wemoCallResponse != null) {
+                                updateState(CHANNEL_STATE, OnOffType.ON);
+                                State newMode = new StringType("Brewing");
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/handler/WemoCoffeeHandler.java]: Could you list all those modes as "stateOptions" in the channel-type definitions?

File: extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/handler/WemoCoffeeHandler.java [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+                                            updateState(CHANNEL_COFFEEMODE, newMode);
+                                            break;
+                                        case "6":
+                                            updateState(CHANNEL_STATE, OnOffType.OFF);
+                                            newMode = new StringType("CleaningBrewing");
+                                            updateState(CHANNEL_COFFEEMODE, newMode);
+                                            break;
+                                        case "7":
+                                            updateState(CHANNEL_STATE, OnOffType.OFF);
+                                            newMode = new StringType("CleaingSoaking");
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/handler/WemoCoffeeHandler.java]: typo?

**Comment by hmerk** [comment]
Thanks for your review @kaikreuzer.
I will check your comments later today. I left those two channels hoping that somebody might come up with an explanation what they indicate and how it should be displayed. I will change the description according to this aim.
Do you have an example for the stateOptions?

File: extensions/binding/org.eclipse.smarthome.binding.wemo/ESH-INF/thing/thing-types.xml [hunk 3]
```diff
@@ -324,7 +324,7 @@
      <channel-type id="brewing" advanced="true">
         <item-type>DateTime</item-type>
         <label>Brewing</label>
-        <description>tba</description>
+        <description>This channel is not documented right now. If You find a valid description, please post to community</description>
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.wemo/ESH-INF/thing/thing-types.xml]: Err. I would suggest to REMOVE the channel from the binding as long as nobody knows what it is good for...

**Comment by kaikreuzer** [comment]
> Do you have an example for the stateOptions?

Most certainly: https://www.eclipse.org/smarthome/documentation/development/bindings/thing-definition.html#state-description

**Pull request merged**